CI fails with

```
Checking code format for regression tests...
nix/conformance/kselftest.nix: not formatted
make[1]: *** [Makefile:174: check] Error 1
```

Not sure if nix scripts have been ever formatted before, but after  running `make format` locally, I have 30 nix files changed:

<img width="1248" height="968" alt="image" src="https://github.com/user-attachments/assets/c2a0e083-35fb-4ac9-a97d-1404f1c54448" />

--- a/test/initramfs/nix/conformance/kselftest.nix
+++ b/test/initramfs/nix/conformance/kselftest.nix
@@ -1,5 +1,8 @@
-{ lib, stdenv, fetchgit, pkgsBuildBuild, pkgsBuildHost, }:
-stdenv.mkDerivation rec {
+{ lib, stdenv, fetchgit, pkgs, pkgsBuildBuild, pkgsBuildHost, }:
+let
+  crossCompilePrefix = pkgsBuildHost.gcc.targetPrefix;
+  hostCc = "${pkgsBuildBuild.gcc}/bin/gcc";
+in stdenv.mkDerivation rec {
   pname = "kselftest";
   version = "6.18";
 
@@ -36,18 +39,17 @@ stdenv.mkDerivation rec {
 
   enableParallelBuilding = true;
 
-  nativeBuildInputs = with pkgsBuildBuild; [ gcc_multi rsync ];
+  nativeBuildInputs = with pkgsBuildBuild; [ rsync ];
 
-  buildInputs = with pkgsBuildHost; [
-    glibc_multi
-    glibc_multi.static
-    libcap.dev
-  ];
+  buildInputs = with pkgs; [ glibc_multi.static libcap.dev ];
 
   buildPhase = ''
     runHook preBuild
-    make ARCH=${stdenv.hostPlatform.linuxArch} -j$NIX_BUILD_CORES headers
+    make ARCH=${stdenv.hostPlatform.linuxArch} \
+         CROSS_COMPILE=${crossCompilePrefix} HOSTCC=${hostCc} \
+         -j$NIX_BUILD_CORES headers
     make -C tools/testing/selftests ARCH=${stdenv.hostPlatform.linuxArch} \
+         CROSS_COMPILE=${crossCompilePrefix} HOSTCC=${hostCc} \
          -j$NIX_BUILD_CORES TARGETS="$kselftestTargets" all
     runHook postBuild
   '';
@@ -55,6 +57,7 @@ stdenv.mkDerivation rec {
   installPhase = ''
     runHook preInstall
     make -C tools/testing/selftests ARCH=${stdenv.hostPlatform.linuxArch} \
+         CROSS_COMPILE=${crossCompilePrefix} HOSTCC=${hostCc} \
          -j$NIX_BUILD_CORES TARGETS="$kselftestTargets" KSFT_INSTALL_PATH=$out install
     runHook postInstall
   '';

